### ISSUE TITLE
Google Analytics cookies still remain after Rejecting All Cookies

### ISSUE THREAD

**Issue by sensibleandy** [manucaralmo/GlowCookies#73]
Hi,

I've added your script to our web site - https://www.sheffieldmca.org.uk

However if I choose to reject all cookies I can still see the GA cookies listed when I reload the pages.

Can you help please?

Andy

**Comment by chrisonline** [comment]
How did you add it?
Can you post the code?

**Comment by chrisonline** [comment]
I have checked your site, you sillt load the GTM code:
![image](https://user-images.githubusercontent.com/2077218/212951077-dfd47b02-2cb2-4a76-af6f-313826033492.png)

**Comment by sensibleandy** [comment]
I don't see the GTA code when I view the source.

Are you looking at the home page of:  https://www.sheffieldmca.org.uk ?

This is the code I added:

<script src="https://cdn.jsdelivr.net/gh/manucaralmo/GlowCookies@3.1.7/src/glowCookies.min.js"></script>
<script>
    glowCookies.start('en', {
        style: 1,
        analytics: 'G-GG3YDDC3KH',
        bannerDescription: "We use our own and third-party cookies to personalise content and to analyse web traffic.",
        hideAfterClick: false,
        policyLink: '/cookies'
    });
</script>

**Comment by chrisonline** [comment]
Sorry, was my mistake.

I have just tested it again and the GTM code does not fire if the cookie banner is shown.
But if I tap on Reject the GTM is fired.

After a reload of the page it works and the GTM code is not fired again.
Strange. Haven't found why this happens.

**Comment by sensibleandy** [comment]
so do you think there's a bug in the GlowCookie system?

How can the GTM code appear if I haven't added it in the first place?

**Comment by chrisonline** [comment]
Yes I think this could be a bug, but I wonder because this does not happen on my website I have added this script.
Any hint: @manucaralmo ?

**Comment by sensibleandy** [comment]
@manucaralmo would you be able to look at https://www.sheffieldmca.org.uk to see if GlowCookie is working for you?

Can you choose to reject all cookies and then see if they are being blocked when you refresh the page?

Thanks

**Comment by chrisonline** [comment]
@sensibleandy After refresh GA is blocked. But after clicking on "reject" GA will be loaded once. 
After refresh, it is not loaded again.

**Comment by sensibleandy** [comment]
Which browser are you using?  Sometimes it seems to work properly and then other times it doesn't work properly.

**Comment by chrisonline** [comment]
I use Chrome browser.

**Comment by ra7khan** [comment]
Hi guys,

I found out where the problem comes from. I had to take a look into the GlowCookies JS.

The problem is caused by the deactivateCookies() function, which is also injecting Analytics. I removed this highlighted line as you can see in the screenshot. After that it works as expected. Tested and it works in both cases (accept/reject).

![glowCookies](https://user-images.githubusercontent.com/61029175/213778704-f760f0e2-0fb3-4230-aa0c-a36c99a64e4b.png)

**Comment by sensibleandy** [comment]
@manucaralmo will you be updating the javascript as a result of what @ra7khan found?

**Comment by ner00** [comment]
Does this even work at all? It keeps letting the cookies be initialized even if I reject and reload. I've also tried commenting the code demonstrated above by @ra7khan. Anyone has a good alternative in mind?

**Comment by sensibleandy** [comment]
I found the solution that civicuk.com provide is the best at managing the cookies

https://www.civicuk.com/cookie-control/download?product=pro

**Comment by z1ad** [comment]
you need to edit glowcookies.js disableTracking() function and add the following code:
`gtag('consent', 'default', {
							  'ad_storage': 'denied',
							  'analytics_storage': 'denied'
						  });`
						  
here is the full function:
```
disableTracking() {
	  // Google Analytics Tracking ('client_storage': 'none')
	  if (this.tracking.AnalyticsCode) {
		let Analytics = document.createElement('script');
		Analytics.setAttribute('src', `https://www.googletagmanager.com/gtag/js?id=${this.tracking.AnalyticsCode}`);
		document.head.appendChild(Analytics);
		let AnalyticsData = document.createElement('script');
		AnalyticsData.text = `window.dataLayer = window.dataLayer || [];
						  function gtag(){dataLayer.push(arguments);}
						  gtag('js', new Date());
						  // from: https://stackoverflow.com/questions/58801416/disabling-cookies-in-google-analytics-gtag-js
						  gtag('consent', 'default', {
							  'ad_storage': 'denied',
							  'analytics_storage': 'denied'
						  });
						  gtag('config', '${this.tracking.AnalyticsCode}' , {
							  'client_storage': 'none',
							  'anonymize_ip': true
						  });`;
		document.head.appendChild(AnalyticsData);
	  }
  
	  // Clear cookies - not working 100%
	  this.clearCookies()
	}
```
ive tested and i can confirm that GA will no longer use a cookie once this is done
